### PR TITLE
Autolev Parser Documentation 2

### PR DESCRIPTION
**Pull request by NikhilPappu**
- other
  - Added the Gotchas section in autolev_parser.rst in the mechanics part of the documentation.

### CODE REVIEW

**Comment by NikhilPappu** [comment]
@moorepants

**Comment by moorepants** [comment]
Merge in master, since we've merged the first PR.

**Comment by NikhilPappu** [comment]
@moorepants This PR is from autolev_parser_docs_2 to master. The first PR didn't get merged so can you please do that?

**Comment by moorepants** [comment]
Oh, I thought I merged it!

**Comment by moorepants** [comment]
Ok, it is merged.

**Comment by NikhilPappu** [comment]
@moorepants Can you now review this PR and the 3rd one #15067 ? Also please take a look at #15013 and #15006 .

**Comment by moorepants** [comment]
Yes, I'll get to them as I have time. I'm aware of all your PRs.

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+     # Note how frame_a is named differently so it doesn't cause a problem.
+     # On the other hand, 'a' gets rewritten from a scalar to a vector.
+     # This should be changed in the future.
+
+
+------------------------------------------------------------------------
+
+- When dealing with Matrices returned by functions, one must check the
+  order of the values as they may not be the same as in Autolev. This is
+  especially the case for eigenvalues and eigenvetors.
```
Comment (Abdullahjavednesar) [inline doc/src/modules/physics/mechanics/autolev_parser.rst]: eigenvetors -> eigenvectors

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+     
+     kd_eqs = [q1d - u1, q2d - u2]
+     kane = me.KanesMethod(frame_n, q_ind=[q1,q2], u_ind=[u1, u2], kd_eqs = kd_eqs)
+     fr, frstar = kane.kanes_equations([particle_p, particle_r], forceList)
+     zero = fr+frstar
+
+------------------------------------------------------------------------
+
+- Need to change ``me.dynamicsymbols._t`` to ``me.dynamicsymbols('t')`` for
+  all occurences of it in the Kane's equations. For example have a look at
```
Comment (Abdullahjavednesar) [inline doc/src/modules/physics/mechanics/autolev_parser.rst]: occurences -> occurrences

**Comment by NikhilPappu** [comment]
@certik 
Travis seems to have some issues. The build passed before but isn't passing now after just fixing typos.

**Comment by certik** [comment]
I restarted the failed builds.

**Review by moorepants — CHANGES_REQUESTED** [review]
This a very helpful document and clearly explained. Nice work. I've left some feedback that I'd like addressed before merging.

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+
+  .. code-block:: python
+  
+     SymPy Code
+     ----------
+     a = q1*a.frame_a.x + q2*frame_a.y
+     b = a.subs({q1:30*0.0174533})
+     # b = a.subs({q1:np.deg2rad(30)}
+
+------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (NikhilPappu) [reply doc/src/modules/physics/mechanics/autolev_parser.rst]: It seems to be valid restructuredText. These lines render well and separate the different sections nicely.

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+     fr, frstar = kane.kanes_equations([particle_p, particle_r], forceList)
+     zero = fr+frstar
+
+------------------------------------------------------------------------
+
+- Need to change ``me.dynamicsymbols._t`` to ``me.dynamicsymbols('t')`` for
+  all occurrences of it in the Kane's equations. For example have a look at
+  line 10 of this `spring damper example <https://github.com/sympy/sympy/blob/master/sympy/parsing/autolev/test_examples/mass_spring_damper.py>`_.
+  This equation is used in forming the Kane's equations so we need to
+  change ``me.dynamicsymbols._t`` to ``me.dynamicsymbols('t')`` in this case.
```
[the remark this answers is not in the document]
Comment (NikhilPappu) [reply doc/src/modules/physics/mechanics/autolev_parser.rst]: The main problem is that if Autolev finds the variable `T` in an equation it substitutes the time instance for it during numerical substitution.
In PyDy, we have to explicitly pass in specifieds in the System class and only these specifieds are substituted for with time instances.
This problem can be fixed if PyDy's System class can substitute the time instances for occurences of dynamicsymbols._t.

**Review by moorepants — APPROVED** [review]

**Pull request merged**